SOC-926 Load localized Facebook JS SDK

armonr opened the pull request:
/CC @lizlux @bkoval @jsutterfield 

Previously, all Facebook plugins on Wikia pages have been rendered in English because the SDK URL had been hardcoded to point to the English version. This changeset aims respect the language of the Wikia the SDK loads for; as a result, all FB plugins on a page are properly localized to the same language.
Facebook documentation on supported languages can be found [here](https://developers.facebook.com/docs/internationalization#locales). Also both country code and language code are required to get the locale right.

Update: Geo cookie is being used now to try and match the user's location with Wikia's language if possible.

For more background info, please see [SOC-926](https://wikia-inc.atlassian.net/browse/SOC-926)

Commit:
SOC-926 Use the localized FB SDK

Commit:
SOC-926 Include facebookLocale in assets

Commit:
SOC-926 Use localized version of FB SDK

Commit:
Use undefined object to clear error

bkoval commented:
Seems like unit tests failure is connected to this pull request, please take a look at the Jenkins build :santa:

bkoval commented on resources/wikia/modules/facebookLocale.js:
```diff
@@ -0,0 +1,159 @@
+		function getSdkUrl(langCode) {
+			var lowerLangCode = langCode.toLowerCase(),
+				countryCode = codes[langCode];
+			if (langCode.length !== 2 || countryCode === undefined) {
```
I think !countryCode would be a sufficient check here, since all the keys in the lookup object are 2 chars long.

bkoval commented on resources/wikia/modules/facebookLocale.js:
```diff
@@ -0,0 +1,159 @@
+	context.Wikia.fbLocale = facebookLocale(context);
+
+	if (context.define && context.define.amd) {
+		context.define('wikia.fbLocale', ['wikia.window'], facebookLocale);
```
Are both Wikia.fbLocale and AMD module needed at the same pageview? Is there a chance both forms will be used on one pageview?

armonr replied:
I'm not quite sure if it is needed, but I tried to follow existing patterns such as https://github.com/Wikia/app/blob/dev/resources/wikia/modules/cookies.js#L146-148. Is there a better way to go about this?

bkoval replied:
I get it. Future-proof JS for the win! Let's leave it as it is :)

bkoval commented:
Minor stuff. Overall it looks nice!

mklucsarits commented:
Aside from failing tests, LGTM

Commit:
SOC-926 Update loader test to mock & include fbLocale
/CC @lizlux

Commit:
Add FB locale module prior to loader.js

Commit:
SOC-926 Also try to match lang to user country for FB SDK

Commit:
Loader styling + fix loader integration test
/CC @lizlux

Commit:
Rename fbLocaleMock to fbLocale in integration test

Commit:
SOC-926 Improve facebookLocale module's readability

Commit:
Remove duplicate inclusion of fbLocale module in assets

lizlux commented:
:+1:

Commit:
SOC-926 /s/wgContentLanguage/wgUserLanguage
wgUserLanguage reflects the user preference language; if not set or anon user, it falls back to wikia language
